### ISSUE TITLE
Extend pie to be able to work with gRPC

### ISSUE THREAD

**Issue by natefinch** [natefinch/pie#10]
[gRPC](http://www.grpc.io/) is really nice and really fast and has a ton of cross platform support (obviously way more than go-compatible RPC).  Enabling gRPC would make a lot of interop problems with pie go away.

**Comment by pires** [comment]
This would be really awesome to have, particularly for stream-based RPC.

**Comment by yanpozka** [comment]
any progress on this ? I was close to create an issue with this topic ;)

**Comment by natefinch** [comment]
No progress, at the time, grpc was still a bit of a moving target. It's nicely released now, so I should look at this again.

**Comment by tgulacsi** [comment]
I've just implemented a gRPC transport for a plugin which I previously used pie (it had some error, which occasionally blocked the connection, which blocked everything).
I'm using Abstract Unix Domain Socket (Linux-specific), as gRPC needs a Listener, a simple connection is not enough (AFAIK).

**Comment by dchenk** [comment]
@tgulacsi and @natefinch can you please help with an answer: Why would you choose communicating over stdin/stdout instead of with Unix domain sockets? Is either method more reliable?

**Comment by tgulacsi** [comment]
No, I think they're the same regarding reliability, just that stdin/stdout
is already there, needs no extra setup (unix domain socket needs to create
a file somewhere, or an abstract socket (Linux specific and less secure))
and more platform-agnostic.

Dmitriy <notifications@github.com> ezt írta (időpont: 2018. aug. 2., Cs,
4:07):

**Comment by natefinch** [comment]
Yeah, mostly because the performance of Unix domain sockets on windows is really bad. ;)
